warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/VirtualIngest.java

```diff
@@ -383,11 +405,58 @@ private void compilePatterns() {
             compiledFieldPatterns = patterns;
         }
         
+        public class VirtualFieldGrouping {
+            private final String group0;
+            private final String group1;
+            
+            public VirtualFieldGrouping(String group0, String group1) {
+                this.group0 = group0;
+                this.group1 = group1;
+            }
+            
+            public String getGroup1() {
+                return group1;
+            }
+            
+            public String getGroup0() {
```

Comment:
Tiny pet peeve. Can we call these something more like getGroup and getSubGroup rather than maintaining an homage to the now defunct String array position in the method name?

Reply:
sure, why not